### PR TITLE
executor: verify the existence of the resource group before switching

### PR DESCRIPTION
**Pull request by JmPotato**
### What problem does this PR solve?

Issue Number: ref #54434.

### What changed and how does it work?

Before switching, verify the existence of the resource group. If the specified resource group does not exist, take no action.

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [x] Manual test

Before:

image

After:

image

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by JmPotato** [comment]
@nolouch @glorv PTAL

**Review by nolouch — COMMENTED** [review]
do we need some tests?

File: pkg/executor/adapter.go [hunk 1]
```diff
@@ -555,7 +555,12 @@ func (a *ExecStmt) Exec(ctx context.Context) (_ sqlexec.RecordSet, err error) {
 			return nil, err
 		}
 		if len(switchGroupName) > 0 {
-			stmtCtx.ResourceGroupName = switchGroupName
+			group, err := rm.ResourceGroupCtl.GetResourceGroup(switchGroupName)
```
Comment (nolouch) [inline pkg/executor/adapter.go:558]: BTW, Have we already checked when setting up a non-existent group?

Comment (JmPotato) [reply pkg/executor/adapter.go:558]: Considering that the records of `QUERY WATCH` and the resource group are decoupled, there is currently no check on the existence of the resource group when setting up. This allows for a more flexible configuration of `QUERY WATCH` and simplifies the code. Here are a few typical scenarios: we can set the corresponding resource group after adding `SWITCH_GROUP`; or after deleting a resource group, the corresponding runaway statements will not fail immediately.

**Comment by JmPotato** [comment]
> do we need some tests?

After trying, I found that the existing unit tests and integration tests could not test this situation well. I first completed the test manually and posted the test results in the PR description. PTAL.

**Review by glorv — APPROVED** [review]

**Review by nolouch — APPROVED** [review]

**Pull request merged**